jupyterlab_sql_editor/ipython_magic/trino/trino.py

```diff
@@ -196,6 +209,47 @@ def display_results(self, results, columns, output, limit=20, show_nonprinting=F
             warnings = []
             json_string = pd.DataFrame.from_records(results, columns=columns).to_json(orient="records")
             json_dict = json.loads(json_string)
+            # cast unsafe ints to str for display
+            for row in json_dict:
+                json_array.append(cast_unsafe_ints_to_str(row, warnings))
+            if show_nonprinting:
+                recursive_escape(json_array)
+            display(warnings, JSON(json_array, expanded=args.expand))
+        elif output == "json_cmt":
+            json_array = []
+            warnings = []
+            json_string = pd.DataFrame.from_records(results, columns=columns).to_json(orient="records")
+            json_dict = json.loads(json_string)
```

Comment:
These two lines can be done in a single step
```suggestion
            json_dict = pd.DataFrame.from_records(results, columns=columns).to_dict(orient="records")
```